File: md_test.js
var MarkdownIt = require('markdown-it');
var fs = require('fs');
var request = require('request');

var config = JSON.parse(fs.readFileSync('config.json', 'utf8'));
var API = config.API;
var SEARCH = API + config.SEARCH;
var LOCAL = config.LOCAL;
var LOCAL_PATH = config.LOCAL_PATH;

var md = new MarkdownIt({
    html: true
});

var processQueries = function(queries){
    queries.map(function(q){
	request.get(q.url, function(err, res, body){
	    var content = JSON.parse(body).wiki_content;
	    var result = md.render(content);    
	    
	   fs.writeFile('markdown-it_results/' + q.id + '.html', result, function(err){
		if (err){
		    console.log(err);
		}
		return;
	    });
	});
    });
};

if (LOCAL){
    var queries = JSON.parse(fs.readFileSync(LOCAL_PATH, 'utf-8'));       
    processQueries(queries);
}
else{
    request.get(SEARCH, function(err, res, body){
	var results = JSON.parse(body).results;
	var queries = results.map(function(r){
	    return {id: r.url.replace(/\//g, ''), url: API + 'project' + r.url + 'wiki/home/content/'};
	});
	processQueries(queries);
    });
}

File: md_test.py
import markdown 
from markdown.extensions import codehilite, fenced_code, wikilinks
import json
from requests import get as rget

config = json.loads(open('config.json', 'r').read())

API = config['API']
SEARCH = API + config['SEARCH']

resp = rget(SEARCH)
results = resp.json().get('results')

queries = map(lambda r: dict(id=r.get('url').replace('/', ''), url=("{0}project{1}wiki/home/content/".format(API, r.get('url')))), results)

'''
        extensions=[
            wikilinks.WikiLinkExtension(
                configs=[
                    ('base_url', ''),
             1       ('end_url', ''),
                    ('build_url', functools.partial(build_wiki_url, node))
                ]
            ),
            fenced_code.FencedCodeExtension(),
            codehilite.CodeHiliteExtension(
                [('css_class', 'highlight')]
            )
         ]
'''

for q in queries:
    resp = rget(q['url'])
    result = markdown.markdown(
        resp.json().get('wiki_content'),
    )    
    
    f = open('markdown-py_results/' + q['id'] + '.html', 'w')
    f.write(result.encode('ascii', 'ignore'))
    f.close()






